app/import_functions_test.go

```diff
@@ -3561,6 +3585,37 @@ func TestImportImportDirectPost(t *testing.T) {
 		assert.Equal(t, post.UserId, th.BasicUser.Id)
 	})
 
+	t.Run("Test with IsPinned", func(t *testing.T) {
+		pinnedValue := true
+		creationTime := model.GetMillis()
+		data := LineImportWorkerData{
+			LineImportData{
+				DirectPost: &DirectPostImportData{
+					ChannelMembers: &[]string{
+						th.BasicUser.Username,
+						th.BasicUser2.Username,
+					},
+					User:     ptrStr(th.BasicUser.Username),
+					Message:  ptrStr("Message with EditAt"),
+					CreateAt: &creationTime,
+					IsPinned: &pinnedValue,
+				},
+			},
+			1,
+		}
+		errLine, err := th.App.importMultipleDirectPostLines(th.Context, []LineImportWorkerData{data}, false)
+		require.Nil(t, err)
+		require.Equal(t, 0, errLine)
+		AssertAllPostsCount(t, th.App, initialPostCount, 8, "")
```

Comment:
Where does the number `8` comes from? Could we do something like `GetPostCount` and then add `1` to it then make the assertion?

Reply:
For this one the number 8 is the change the in the posts number, I have been following the other tests pattern in the `TestImportImportDirectPost` function.
For example the the first and the second tests are trying to import a post (valid and invalid) in dry run mode, and the third test is trying to add an invalid post in apply mode, so you'll see that all of them are doing
`AssertAllPostsCount(t, th.App, initialPostCount, 0, "")`.
And in the fourth tests you'll see that the changed posts number will start increasing to 1 and in the fifth it will be updating a previous post so it will stay 1, and in the sixth will be 2 and so on, each time a post is added the changed number of posts will increase by 1.
this is kinda hard to test with a `len()` or `GetPostCount` function because we don't have something like an `addedPosts` list and have it updated, and I was trying to follow the test style without changing it.

Reply:
@isacikgoz Other than that, I have done all the edits you suggested and it should be good now.